src/js/page/results-cache.js

```diff
@@ -1,3 +1,5 @@
+// TODO: switch to Map/Set?
```

Comment:
Yeah, a map would work well here, and the code would likely read better. Doesn't need to be done for this PR tho